iso15118/shared/messages/iso15118_20/common_types.py

```diff
@@ -174,6 +174,125 @@ class RationalNumber(BaseModel):
+        if float_value is None:
+            return None
+        exponent, value = cls._convert_to_exponent_number(float_value)
+        return RationalNumber(exponent=exponent, value=value)
+
+    @classmethod
+    def _convert_to_exponent_number(
+        cls,
+        float_value: float,
+        min_limit: float = INT_16_MIN,
+        max_limit: float = INT_16_MAX,
+    ) -> Tuple[int, int]:
+        """
+        Convert to exponent number, by finding the best fit exponent.
+
+        For example, max limit for PVEVSEMinCurrentLimit is 400
+        value = 0.0000234 => exponent = -3, return value = 0
+        value = 0.000234 => exponent = -3, return value = 0
+        value = 0.00234 => exponent = -3, return value = 2
+        value = 0.0234 => exponent = -3, return value = 23
+        value = 0.234 => exponent = -3, return value = 234
+        value = 2.34 => exponent = -2, return value = 234
+        value = 23.4 => exponent = -1, return value = 234
+        value = 234 => exponent = 0, return value = 234
+        value = 2340 => exponent = 1, return value = 234
+        value = 23400 => exponent = 2, return value = 234
+        value = 234000 => exponent = 3, return value = 234
+
+        value = 0.4 => exponent = -3, return value = 400
+        value = 0.356 => exponent = -3, return value = 356
+        value = 0.157 => exponent = -3, return value = 157
+        value = 0.00356 => exponent = -3, return value = 3
+        value = 0.634 => exponent = -2, return value = 63
+
+        Note:
+            In ISO 15118-2, the min limit is 0 and the max is
+            dependent on the field.
+            In ISO 15118-20, the limits are defined by the
+            RationalNumber type which are the limits of the
+            int16 range: [-32768, 32767].
+            Also, in ISO 15118-20, the exponent assumes the range
+            of [-128, 127], but currently that is not being
+            taken into consideration.
+        Args:
+            value (float): parameter value
+
+        Returns:
+           a tuple where the first item is the exponent and other is the
+           int value with the applied exponent
+        """
+        if float_value < 0 and min_limit == 0:
+            raise ValueError(
+                "The conversion of negative values "
+                "for a range with a min limit of 0, is invalid!"
+            )
+        # This is to ensure that we keep the value to be converted, within
+        # the range of the classes RationalNumber and PhysicalValue 'value' data type
+        # int16
+        min_limit = max(min_limit, INT_16_MIN)
+        # For example, the class PVEVSEMaxPowerLimit has a _max_limit = 200000,
+        # so, if the value to be converted is 200000, we cant represent it
+        # with exponent = 0; instead, we need to take into account
+        # the INT_16_MAX = 32768 and represent the value as
+        # value = 20000, exponent = 1
+        max_limit = min(max_limit, INT_16_MAX)
```

Comment:
we will have trouble with my code in https://github.com/SwitchEV/iso15118/pull/319/files#diff-dffae5111fad8ea791bc869910a9e31c6cd21e49ab7c89bc7bc6b6ad36a8ab2bR51

I guess it is better if I set the limits to the max representations of it in my code

Reply:
Hmm..this is however specific to -20. The change you added would apply only to -2 and din, right?

Reply:
that is true...we dont use the convert_to_exponent number there do we?
not sure why we dont tho lol